src/python/pants/backend/python/util_rules/pex_test_utils.py

```diff
@@ -69,59 +69,38 @@ class PexData:
 
 
 def get_all_data(rule_runner: RuleRunner, pex: Pex | VenvPex) -> PexData:
+    # We fish PEX-INFO out of the pex manually rather than running PEX_TOOLS, as
+    # we don't know if the pex can run on the current system.
     if isinstance(pex, VenvPex):
         digest = pex.digest
         sandbox_path = pex.pex_filename
-        process = rule_runner.request(
-            Process,
-            [
-                VenvPexProcess(
-                    pex,
-                    argv=["info"],
-                    extra_env=dict(PEX_TOOLS="1"),
-                    description="Extract PEX-INFO.",
-                ),
-            ],
-        )
     else:
         digest = pex.digest
         sandbox_path = pex.name
-        pex_pex = rule_runner.request(PexPEX, [])
-        process = rule_runner.request(
-            Process,
-            [
-                PexProcess(
-                    Pex(digest=pex_pex.digest, name=pex_pex.exe, python=pex.python),
-                    argv=["-m", "pex.tools", pex.name, "info"],
-                    input_digest=pex.digest,
-                    extra_env=dict(PEX_INTERPRETER="1"),
-                    description="Extract PEX-INFO.",
-                )
-            ],
-        )
 
     rule_runner.scheduler.write_digest(digest)
-    local_path = PurePath(rule_runner.build_root) / "test.pex"
-    result = rule_runner.request(ProcessResult, [process])
-    pex_info_content = result.stdout.decode()
+    local_path = PurePath(rule_runner.build_root) / sandbox_path
```

Comment:
This change uncovered a bug here - we hardcoded "test.pex" as the name, which was not true for all tests that used this util, but the ones that didn't happened to not care about the "files" list, and so didn't notice that it was empty...